### ISSUE TITLE
[Bug]: Authentication is closing session when its not supposed to

### ISSUE THREAD

**Issue by dpfaffenbauer** [pimcore/pimcore#11927]
### Expected behavior

`Authentication::authenticateSession($request);` should find the current Admin User in the Session

### Actual behavior

`Authentication::authenticateSession($request);` finds the current user, but it uses `AdminSessionHandler::useSessionAttributeBag` which runs `writeClose` closing the current session and therefore hindering Symfony from sending the right newly, generated and migrated, session ID to the user since the session is actually closed.

### Steps to reproduce

Enabled Pimcore Targeting:

```
pimcore:
    targeting:
        enabled: false
```

And authenticate to a firewall. In my case it happens with CoreShop Tests: https://github.com/coreshop/CoreShop/pull/1970/files#diff-2f8cab0f04dcf2454dbe1699f7f35fd8d169952ad38ba6baee510b20bde8ef6cR5

**Comment by brusch** [comment]
Fixed by #12022

**Comment by dpfaffenbauer** [comment]
💪

**Comment by solverat** [comment]
@dvesh3, @brusch: Now this leads to issues in our test cases. Basically, every functional test will fail, as long the targeting feature is enabled.

![image](https://user-images.githubusercontent.com/700119/202797590-b8823537-3760-4bab-a16a-7d20d3d1a93f.png)


The reason for that is this line:
https://github.com/pimcore/pimcore/blob/52429018731f731544b176954f9545f4c2e16ff1/bundles/AdminBundle/Session/Handler/AdminSessionHandler.php#L300

In a perfect World, the request should reach `AdminSessionHandler`, which has been initally started here: 

https://github.com/pimcore/pimcore/blob/c7f023cf2e349bba8794795f6365447b836c887a/lib/Targeting/EventListener/ToolbarListener.php#L160

I don't know if that's possible. However, it would be great, if the request fetch is allowed to fail silently. Something like this:

```php
private function shouldWriteAndClose(): bool
{
    try {
        $request = $this->requestHelper->getMainRequest();
    } catch (\Throwable $e) {
        return true;
    }
    
    return $this->canWriteAndClose ??= $this->isAdminRequest($request);
}
```

Backlink: #13614 

***

<details>
<summary>FullStack Trace</summary>

```
LogicException:
There is no main request available.

  at vendor/pimcore/pimcore/lib/Http/RequestHelper.php:129
  at Pimcore\Http\RequestHelper->getMainRequest()
     (vendor/pimcore/pimcore/bundles/AdminBundle/Session/Handler/AdminSessionHandler.php:300)
  at Pimcore\Bundle\AdminBundle\Session\Handler\AdminSessionHandler->shouldWriteAndClose()
     (vendor/pimcore/pimcore/bundles/AdminBundle/Session/Handler/AdminSessionHandler.php:275)
  at Pimcore\Bundle\AdminBundle\Session\Handler\AdminSessionHandler->writeClose()
     (vendor/pimcore/pimcore/bundles/AdminBundle/Session/Handler/AdminSessionHandler.php:133)
  at Pimcore\Bundle\AdminBundle\Session\Handler\AdminSessionHandler->useSessionAttributeBag(object(Closure), 'pimcore_admin')
     (vendor/pimcore/pimcore/bundles/AdminBundle/Session/Handler/AdminSessionHandler.php:146)
  at Pimcore\Bundle\AdminBundle\Session\Handler\AdminSessionHandler->getReadOnlyAttributeBag('pimcore_admin')
     (vendor/pimcore/pimcore/lib/Tool/Session.php:135)
  at Pimcore\Tool\Session::getReadOnly()
     (vendor/pimcore/pimcore/lib/Tool/Authentication.php:70)
  at Pimcore\Tool\Authentication::authenticateSession(object(Request))
     (vendor/pimcore/pimcore/lib/Targeting/EventListener/ToolbarListener.php:160)
  at Pimcore\Targeting\EventListener\ToolbarListener->requestCanDebug(object(Request))
     (vendor/pimcore/pimcore/lib/Targeting/EventListener/ToolbarListener.php:127)
  at Pimcore\Targeting\EventListener\ToolbarListener->onKernelResponse(object(ResponseEvent), 'kernel.response', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/Debug/WrappedListener.php:117)
  at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(ResponseEvent), 'kernel.response', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:230)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.response', object(ResponseEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:59)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(ResponseEvent), 'kernel.response')
     (vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:154)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(ResponseEvent), 'kernel.response')
     (vendor/symfony/http-kernel/HttpKernel.php:196)
  at Symfony\Component\HttpKernel\HttpKernel->filterResponse(object(Response), object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:184)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:75)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/HttpKernelBrowser.php:65)
  at Symfony\Component\HttpKernel\HttpKernelBrowser->doRequest(object(Request))
     (vendor/codeception/module-symfony/src/Codeception/Lib/Connector/Symfony.php:73)
  at Codeception\Lib\Connector\Symfony->doRequest(object(Request))
     (vendor/symfony/browser-kit/AbstractBrowser.php:398)
  at Symfony\Component\BrowserKit\AbstractBrowser->request('POST', 'http://localhost/form-test', array('formbuilder_1' => array('formId' => '1', 'formCl' => '[]', 'inputUserName' => '', 'formRuntimeData' => '{"form_preset":"custom","form_template":"form_div_layout","form_output_workflow":1,"custom_options":[]}', 'formRuntimeDataToken' => '12e9b197d47466234c25686.9vZTvC-xzR1_Go3XneRD3KbnIoyAnGodBAsVnfY6ScA.tcEy0h_-_HATdLW8qp0M7tOhZcL5pRtQNV8n-IRDf7i6ozaReIL9LE5r3A', 'text' => '5', '_token' => 'a2cebb9e3.SGLTpW0kyzpR2QPGYCI1TFmwriXBoZEIEc35pJ5Yq3A.eQux31lDg0MzhlekKmBUCxCA4mHz9alNZ5m4684270k-IIndVUr8QgW8Nw', 'submit' => '')), array(), array('HTTP_USER_AGENT' => 'Symfony BrowserKit', 'REQUEST_TIME' => 1668803206, 'REQUEST_TIME_FLOAT' => 1668803206.829057, 'HTTP_REFERER' => 'http://localhost/form-test', 'HTTP_HOST' => 'localhost', 'HTTPS' => false), 'formbuilder_1[formId]=1&formbuilder_1[formCl]=[]&formbuilder_1[inputUserName]=&formbuilder_1[formRuntimeData]={"form_preset":"custom","form_template":"form_div_layout","form_output_workflow":1,"custom_options":[]}&formbuilder_1[formRuntimeDataToken]=12e9b197d47466234c25686.9vZTvC-xzR1_Go3XneRD3KbnIoyAnGodBAsVnfY6ScA.tcEy0h_-_HATdLW8qp0M7tOhZcL5pRtQNV8n-IRDf7i6ozaReIL9LE5r3A&formbuilder_1[text]=5&formbuilder_1[_token]=a2cebb9e3.SGLTpW0kyzpR2QPGYCI1TFmwriXBoZEIEc35pJ5Yq3A.eQux31lDg0MzhlekKmBUCxCA4mHz9alNZ5m4684270k-IIndVUr8QgW8Nw&formbuilder_1[submit]=', true)
     (vendor/codeception/lib-innerbrowser/src/Codeception/Lib/InnerBrowser.php:248)
  at Codeception\Lib\InnerBrowser->clientRequest('POST', 'http://localhost/form-test', array('formbuilder_1' => array('formId' => '1', 'formCl' => '[]', 'inputUserName' => '', 'formRuntimeData' => '{"form_preset":"custom","form_template":"form_div_layout","form_output_workflow":1,"custom_options":[]}', 'formRuntimeDataToken' => '12e9b197d47466234c25686.9vZTvC-xzR1_Go3XneRD3KbnIoyAnGodBAsVnfY6ScA.tcEy0h_-_HATdLW8qp0M7tOhZcL5pRtQNV8n-IRDf7i6ozaReIL9LE5r3A', 'text' => '5', '_token' => 'a2cebb9e3.SGLTpW0kyzpR2QPGYCI1TFmwriXBoZEIEc35pJ5Yq3A.eQux31lDg0MzhlekKmBUCxCA4mHz9alNZ5m4684270k-IIndVUr8QgW8Nw', 'submit' => '')), array())
     (vendor/codeception/lib-innerbrowser/src/Codeception/Lib/InnerBrowser.php:943)
  at Codeception\Lib\InnerBrowser->proceedSubmitForm(object(Crawler), array('formbuilder_1[submit]' => ''))
     (vendor/codeception/lib-innerbrowser/src/Codeception/Lib/InnerBrowser.php:497)
  at Codeception\Lib\InnerBrowser->clickButton(object(DOMElement))
     (vendor/codeception/lib-innerbrowser/src/Codeception/Lib/InnerBrowser.php:461)
  at Codeception\Lib\InnerBrowser->clickByLocator('form[name*="formbuilder_1"]  button#formbuilder_1_submit')
     (vendor/codeception/lib-innerbrowser/src/Codeception/Lib/InnerBrowser.php:434)
  at Codeception\Lib\InnerBrowser->click('form[name*="formbuilder_1"]  button#formbuilder_1_submit')
  at call_user_func_array(array(object(PimcoreCore), 'click'), array('form[name*="formbuilder_1"]  button#formbuilder_1_submit'))
     (vendor/codeception/codeception/src/Codeception/Step.php:285)
  at Codeception\Step->run(object(ModuleContainer))
     (vendor/codeception/codeception/src/Codeception/Scenario.php:76)
  at Codeception\Scenario->runStep(object(Action))
     (lib/test-bundle/tests/_support/_generated/FunctionalTesterActions.php:1383)
  at DachcomBundle\Test\FunctionalTester->click('form[name*="formbuilder_1"]  button#formbuilder_1_submit')
     (lib/test-bundle/tests/functional/Constraints/RangeConstraintCest.php:84)
  at DachcomBundle\Test\functional\Constraints\RangeConstraintCest->testRangeConstraintWithAllFieldsAndMinValue(object(FunctionalTester))
  at ReflectionMethod->invokeArgs(object(RangeConstraintCest), array(object(FunctionalTester)))
     (vendor/codeception/codeception/src/Codeception/Lib/Di.php:128)
  at Codeception\Lib\Di->injectDependencies(object(RangeConstraintCest), 'testRangeConstraintWithAllFieldsAndMinValue', array(object(FunctionalTester), object(Scenario)))
     (vendor/codeception/codeception/src/Codeception/Test/Cest.php:138)
  at Codeception\Test\Cest->invoke('testRangeConstraintWithAllFieldsAndMinValue', array(object(FunctionalTester), object(Scenario)))
     (vendor/codeception/codeception/src/Codeception/Test/Cest.php:150)
  at Codeception\Test\Cest->executeTestMethod(object(FunctionalTester))
     (vendor/codeception/codeception/src/Codeception/Test/Cest.php:82)
  at Codeception\Test\Cest->test()
     (vendor/codeception/codeception/src/Codeception/Test/Test.php:98)
  at Codeception\Test\Test->run(object(TestResult))
     (vendor/phpunit/phpunit/src/Framework/TestSuite.php:673)
  at PHPUnit\Framework\TestSuite->run(object(TestResult))
     (vendor/codeception/phpunit-wrapper/src/Runner.php:117)
  at Codeception\PHPUnit\Runner->doEnhancedRun(object(Suite), object(TestResult), array('silent' => false, 'debug' => true, 'steps' => false, 'html' => false, 'xml' => false, 'phpunit-xml' => false, 'no-redirect' => true, 'json' => false, 'tap' => false, 'report' => false, 'colors' => true, 'coverage' => false, 'coverage-xml' => false, 'coverage-html' => false, 'coverage-text' => false, 'coverage-crap4j' => false, 'coverage-cobertura' => false, 'coverage-phpunit' => false, 'groups' => array(), 'excludeGroups' => array(), 'filter' => 'testRangeConstraintWithAllFieldsAndMinValue', 'env' => array('local'), 'fail-fast' => false, 'ansi' => true, 'verbosity' => 128, 'interactive' => true, 'no-rebuild' => false, 'quiet' => false, 'bootstrap' => '_bootstrap.php', 'memory_limit' => -1, 'strict_xml' => false, 'lint' => true, 'backup_globals' => true, 'log_incomplete_skipped' => false, 'report_useless_tests' => false, 'disallow_test_output' => false, 'be_strict_about_changes_to_global_state' => false, 'shuffle' => false, 'seed' => 1084748837, 'listeners' => array(object(UI)), 'addUncoveredFilesFromWhitelist' => true, 'backupGlobals' => null, 'backupStaticAttributes' => null, 'beStrictAboutChangesToGlobalState' => null, 'beStrictAboutResourceUsageDuringSmallTests' => false, 'cacheResult' => true, 'cacheTokens' => false, 'columns' => 80, 'convertDeprecationsToExceptions' => true, 'convertErrorsToExceptions' => true, 'convertNoticesToExceptions' => true, 'convertWarningsToExceptions' => true, 'crap4jThreshold' => 30, 'disallowTestOutput' => false, 'disallowTodoAnnotatedTests' => false, 'defaultTimeLimit' => 0, 'enforceTimeLimit' => false, 'failOnRisky' => false, 'failOnWarning' => false, 'executionOrderDefects' => 0, 'processIsolation' => false, 'processUncoveredFilesFromWhitelist' => false, 'randomOrderSeed' => 1668803157, 'registerMockObjectsFromTestArgumentsRecursively' => false, 'repeat' => false, 'reportHighLowerBound' => 90, 'reportLowUpperBound' => 50, 'reportUselessTests' => true, 'reverseList' => false, 'executionOrder' => 0, 'resolveDependencies' => true, 'stopOnError' => false, 'stopOnFailure' => false, 'stopOnIncomplete' => false, 'stopOnRisky' => false, 'stopOnSkipped' => false, 'stopOnWarning' => false, 'stopOnDefect' => false, 'strictCoverage' => false, 'testdoxExcludeGroups' => array(), 'testdoxGroups' => array(), 'timeoutForLargeTests' => 60, 'timeoutForMediumTests' => 10, 'timeoutForSmallTests' => 1, 'verbose' => false))
     (vendor/codeception/codeception/src/Codeception/SuiteManager.php:161)
  at Codeception\SuiteManager->run(object(Runner), object(TestResult), array('silent' => false, 'debug' => true, 'steps' => false, 'html' => false, 'xml' => false, 'phpunit-xml' => false, 'no-redirect' => true, 'json' => false, 'tap' => false, 'report' => false, 'colors' => true, 'coverage' => false, 'coverage-xml' => false, 'coverage-html' => false, 'coverage-text' => false, 'coverage-crap4j' => false, 'coverage-cobertura' => false, 'coverage-phpunit' => false, 'groups' => array(), 'excludeGroups' => array(), 'filter' => 'testRangeConstraintWithAllFieldsAndMinValue', 'env' => array('local'), 'fail-fast' => false, 'ansi' => true, 'verbosity' => 128, 'interactive' => true, 'no-rebuild' => false, 'quiet' => false, 'bootstrap' => '_bootstrap.php', 'memory_limit' => -1, 'strict_xml' => false, 'lint' => true, 'backup_globals' => true, 'log_incomplete_skipped' => false, 'report_useless_tests' => false, 'disallow_test_output' => false, 'be_strict_about_changes_to_global_state' => false, 'shuffle' => false, 'seed' => 1084748837, 'listeners' => array(), 'addUncoveredFilesFromWhitelist' => true, 'backupGlobals' => null, 'backupStaticAttributes' => null, 'beStrictAboutChangesToGlobalState' => null, 'beStrictAboutResourceUsageDuringSmallTests' => false, 'cacheResult' => true, 'cacheTokens' => false, 'columns' => 80, 'convertDeprecationsToExceptions' => true, 'convertErrorsToExceptions' => true, 'convertNoticesToExceptions' => true, 'convertWarningsToExceptions' => true, 'crap4jThreshold' => 30, 'disallowTestOutput' => false, 'disallowTodoAnnotatedTests' => false, 'defaultTimeLimit' => 0, 'enforceTimeLimit' => false, 'failOnRisky' => false, 'failOnWarning' => false, 'executionOrderDefects' => 0, 'processIsolation' => false, 'processUncoveredFilesFromWhitelist' => false, 'randomOrderSeed' => 1668803157, 'registerMockObjectsFromTestArgumentsRecursively' => false, 'repeat' => false, 'reportHighLowerBound' => 90, 'reportLowUpperBound' => 50, 'reportUselessTests' => true, 'reverseList' => false, 'executionOrder' => 0, 'resolveDependencies' => true, 'stopOnError' => false, 'stopOnFailure' => false, 'stopOnIncomplete' => false, 'stopOnRisky' => false, 'stopOnSkipped' => false, 'stopOnWarning' => false, 'stopOnDefect' => false, 'strictCoverage' => false, 'testdoxExcludeGroups' => array(), 'testdoxGroups' => array(), 'timeoutForLargeTests' => 60, 'timeoutForMediumTests' => 10, 'timeoutForSmallTests' => 1, 'verbose' => false))
     (vendor/codeception/codeception/src/Codeception/Codecept.php:208)
  at Codeception\Codecept->runSuite(array('modules' => array('config' => array('\Dachcom\Codeception\Helper\Browser\WebDriver' => array('browser' => 'chrome', 'port' => 4444, 'restart' => true, 'wait' => 1, 'capabilities' => array('chromeOptions' => array('args' => array('--no-sandbox', '--disable-extensions', '--headless', '--disable-gpu', '--disable-dev-shm-usage', '--window-size=1280,1024'))))), 'enabled' => array(array('\Dachcom\Codeception\Helper\PimcoreCore' => array('connect_db' => true, 'rebootable_client' => true)), array('\Dachcom\Codeception\Helper\PimcoreBundleCore' => array('run_installer' => true)), array('\Dachcom\Codeception\Helper\Browser\PhpBrowser' => array('depends' => '\\Dachcom\\Codeception\\Helper\\PimcoreCore')), array('\Dachcom\Codeception\Helper\PimcoreAdminJson' => array('depends' => '\\Dachcom\\Codeception\\Helper\\PimcoreCore')), array('\Dachcom\Codeception\Helper\PimcoreAdminCsv' => array('depends' => '\\Dachcom\\Codeception\\Helper\\PimcoreCore')), '\\Dachcom\\Codeception\\Helper\\PimcoreUser', '\\DachcomBundle\\Test\\Helper\\PimcoreBackend'), 'depends' => array()), 'env' => array('github' => array('modules' => array('config' => array('\Dachcom\Codeception\Helper\Browser\WebDriver' => array('browser' => 'chrome', 'port' => 9515, 'restart' => true, 'wait' => 1, 'capabilities' => array('chromeOptions' => array('args' => array('--no-sandbox', '--disable-extensions', '--headless', '--disable-gpu', '--disable-dev-shm-usage', '--window-size=1280,1024'), 'prefs' => array('download.default_directory' => '/var/www/html/public/lib/test-bundle/tests/_data/downloads'))))))), 'local' => array('modules' => array('config' => array('\Dachcom\Codeception\Helper\Browser\WebDriver' => array('browser' => 'chrome', 'port' => 4444, 'restart' => true, 'wait' => 1, 'capabilities' => array('chromeOptions' => array('args' => array('--no-sandbox', '--disable-extensions', '--headless', '--disable-gpu', '--disable-dev-shm-usage', '--window-size=1280,1024')))))))), 'actor' => 'FunctionalTester', 'error_level' => 'getenv("PIMCORE_PHP_ERROR_REPORTING")', 'bootstrap' => '_bootstrap.php', 'memory_limit' => -1, 'colors' => true, 'strict_xml' => false, 'lint' => true, 'backup_globals' => true, 'log_incomplete_skipped' => false, 'report_useless_tests' => false, 'disallow_test_output' => false, 'be_strict_about_changes_to_global_state' => false, 'shuffle' => false, 'coverage' => array(), 'namespace' => 'DachcomBundle\\Test', 'groups' => array(), 'gherkin' => array(), 'extensions' => array('enabled' => array('Codeception\\Extension\\RunFailed'), 'config' => array(), 'commands' => array()), 'class_name' => null, 'step_decorators' => 'Codeception\\Step\\ConditionalAssertion', 'path' => '/var/www/html/public/lib/test-bundle/tests/functional/', 'extends' => null, 'formats' => array(), 'convert_deprecations_to_exceptions' => false, 'current_environment' => 'local'), 'functional (local)', 'Constraints/RangeConstraintCest.php')
     (vendor/codeception/codeception/src/Codeception/Codecept.php:197)
  at Codeception\Codecept->run('functional', 'Constraints/RangeConstraintCest.php', array('env' => array('github' => array('modules' => array('config' => array('\Dachcom\Codeception\Helper\Browser\WebDriver' => array('browser' => 'chrome', 'port' => 9515, 'restart' => true, 'wait' => 1, 'capabilities' => array('chromeOptions' => array('args' => array('--no-sandbox', '--disable-extensions', '--headless', '--disable-gpu', '--disable-dev-shm-usage', '--window-size=1280,1024'), 'prefs' => array('download.default_directory' => '/var/www/html/public/lib/test-bundle/tests/_data/downloads'))))))), 'local' => array('modules' => array('config' => array('\Dachcom\Codeception\Helper\Browser\WebDriver' => array('browser' => 'chrome', 'port' => 4444, 'restart' => true, 'wait' => 1, 'capabilities' => array('chromeOptions' => array('args' => array('--no-sandbox', '--disable-extensions', '--headless', '--disable-gpu', '--disable-dev-shm-usage', '--window-size=1280,1024')))))))), 'actor' => 'FunctionalTester', 'error_level' => 'getenv("PIMCORE_PHP_ERROR_REPORTING")', 'modules' => array('enabled' => array(array('\Dachcom\Codeception\Helper\PimcoreCore' => array('connect_db' => true, 'rebootable_client' => true)), array('\Dachcom\Codeception\Helper\PimcoreBundleCore' => array('run_installer' => true)), array('\Dachcom\Codeception\Helper\Browser\PhpBrowser' => array('depends' => '\\Dachcom\\Codeception\\Helper\\PimcoreCore')), array('\Dachcom\Codeception\Helper\PimcoreAdminJson' => array('depends' => '\\Dachcom\\Codeception\\Helper\\PimcoreCore')), array('\Dachcom\Codeception\Helper\PimcoreAdminCsv' => array('depends' => '\\Dachcom\\Codeception\\Helper\\PimcoreCore')), '\\Dachcom\\Codeception\\Helper\\PimcoreUser', '\\DachcomBundle\\Test\\Helper\\PimcoreBackend'), 'config' => array(), 'depends' => array()), 'bootstrap' => '_bootstrap.php', 'memory_limit' => -1, 'colors' => true, 'strict_xml' => false, 'lint' => true, 'backup_globals' => true, 'log_incomplete_skipped' => false, 'report_useless_tests' => false, 'disallow_test_output' => false, 'be_strict_about_changes_to_global_state' => false, 'shuffle' => false, 'coverage' => array(), 'namespace' => 'DachcomBundle\\Test', 'groups' => array(), 'gherkin' => array(), 'extensions' => array('enabled' => array('Codeception\\Extension\\RunFailed'), 'config' => array(), 'commands' => array()), 'class_name' => null, 'step_decorators' => 'Codeception\\Step\\ConditionalAssertion', 'path' => '/var/www/html/public/lib/test-bundle/tests/functional/', 'extends' => null, 'formats' => array(), 'convert_deprecations_to_exceptions' => false))
     (vendor/codeception/codeception/src/Codeception/Command/Run.php:401)
  at Codeception\Command\Run->execute(object(ArgvInput), object(ConsoleOutput))
     (vendor/symfony/console/Command/Command.php:298)
  at Symfony\Component\Console\Command\Command->run(object(ArgvInput), object(ConsoleOutput))
     (vendor/symfony/console/Application.php:1040)
  at Symfony\Component\Console\Application->doRunCommand(object(Run), object(ArgvInput), object(ConsoleOutput))
     (vendor/symfony/console/Application.php:301)
  at Symfony\Component\Console\Application->doRun(object(ArgvInput), object(ConsoleOutput))
     (vendor/symfony/console/Application.php:171)
  at Symfony\Component\Console\Application->run(object(ArgvInput), object(ConsoleOutput))
     (vendor/codeception/codeception/src/Codeception/Application.php:117)
  at Codeception\Application->run()
     (vendor/codeception/codeception/app.php:46)
  at {closure}()
     (vendor/codeception/codeception/app.php:47)
  at require('/var/www/html/public/vendor/codeception/codeception/app.php')
     (vendor/codeception/codeception/codecept:7)
  at include('/var/www/html/public/vendor/codeception/codeception/codecept')
     (vendor/bin/codecept:120)
```
</details>


***

**Comment by brusch** [comment]
@solverat thanks for your feedback! 
What if we only allow it to fail silently when `'cli' === PHP_SAPI` ? 

@dvesh3 wdyt?

**Comment by solverat** [comment]
@brusch the longer I look at the method `shouldWriteAndClose` the more I dislike it. :) 

However, I've just tested it with your proposed `cli` check, and it works too:

```php
private function shouldWriteAndClose(): bool
{
    try {
        $request = $this->requestHelper->getMainRequest();
    } catch (\Throwable $e) {

        if (PHP_SAPI !== 'cli') {
            throw $e;
        }

        return true;
    }

    return $this->canWriteAndClose ??= $this->isAdminRequest($request);
}
```

**Comment by brusch** [comment]
@solverat thanks for your feedback! 
So if there are no more better ideas I'd go for this for now. @dvesh3

**Comment by dvesh3** [comment]
I would not go for skip in cli mode as you could have tests with dummy requests and session (like `buildSession()` here https://github.com/pimcore/pimcore/blob/c1f88e07434e56a1c04310cf6ec0588e8bce3687/tests/_support/Test/ModelTestCase.php#L64 

Please see suggested fix https://github.com/pimcore/pimcore/pull/13637

**Comment by kingjia90** [comment]
Closing again, fixed by https://github.com/pimcore/pimcore/pull/13637

**Comment by Cruiser13** [comment]
Is this fixed in pimcore v10.5.16 @kingjia90 ? If we setup a frontend firewall (like https://github.com/pimcore/demo/blob/11.x/config/packages/security.yaml#L58 ) and have a code like this in our controller:
`$user = Authentication::authenticateSession($request);`
We'll get logged out from pimcore admin backend on each request.

**Comment by kingjia90** [comment]
> Is this fixed in pimcore v10.5.16 @kingjia90 ? If we setup a frontend firewall (like https://github.com/pimcore/demo/blob/11.x/config/packages/security.yaml#L58 ) and have a code like this in our controller: `$user = Authentication::authenticateSession($request);` We'll get logged out from pimcore admin backend on each request.

It seems working on my end, maybe i am missing something, what do you mean for "we'll get logged out on each request"? where are the authenticateSession done? if possible, an example on how to reproduce it would be even better

**Comment by Cruiser13** [comment]
@kingjia90 we have this in our controllers (e.g. DefaultController -> defaultAction) to check if a pimcore admin is logged in.
